x/wasm/internal/keeper/reflect_test.go

```diff
@@ -296,6 +297,79 @@ func TestMaskReflectCustomQuery(t *testing.T) {
 	assert.Equal(t, resp.Text, "ALL CAPS NOW")
 }
 
+func TestReflectStargateQuery(t *testing.T) {
+	cdc := MakeTestCodec(t)
+	ctx, keepers := CreateTestInput(t, false, MaskFeatures, maskEncoders(cdc), maskPlugins())
+	accKeeper, keeper, bankKeeper := keepers.AccountKeeper, keepers.WasmKeeper, keepers.BankKeeper
+
+	funds := sdk.NewCoins(sdk.NewInt64Coin("denom", 320000))
+	contractStart := sdk.NewCoins(sdk.NewInt64Coin("denom", 40000))
+	expectedBalance := funds.Sub(contractStart)
+	creator := createFakeFundedAccount(t, ctx, accKeeper, bankKeeper, funds)
+
+	// upload code
+	maskCode, err := ioutil.ReadFile("./testdata/reflect.wasm")
+	require.NoError(t, err)
+	codeID, err := keeper.Create(ctx, creator, maskCode, "", "", nil)
+	require.NoError(t, err)
+	require.Equal(t, uint64(1), codeID)
+
+	// creator instantiates a contract and gives it tokens
+	contractAddr, _, err := keeper.Instantiate(ctx, codeID, creator, nil, []byte("{}"), "mask contract 1", contractStart)
+	require.NoError(t, err)
+	require.NotEmpty(t, contractAddr)
+
+	// first, normal query for the bank balance (to make sure our query is proper)
+	bankQuery := wasmvmtypes.QueryRequest{
+		Bank: &wasmvmtypes.BankQuery{
+			AllBalances: &wasmvmtypes.AllBalancesQuery{
+				Address: creator.String(),
+			},
+		},
+	}
+	simpleQueryBz, err := json.Marshal(MaskQueryMsg{
+		Chain: &ChainQuery{Request: &bankQuery},
+	})
+	require.NoError(t, err)
+	simpleRes, err := keeper.QuerySmart(ctx, contractAddr, simpleQueryBz)
+	require.NoError(t, err)
+	var simpleChain ChainResponse
+	mustParse(t, simpleRes, &simpleChain)
+	var simpleBalance wasmvmtypes.AllBalancesResponse
+	mustParse(t, simpleChain.Data, &simpleBalance)
+	require.Equal(t, len(expectedBalance), len(simpleBalance.Amount))
+	assert.Equal(t, simpleBalance.Amount[0].Amount, expectedBalance[0].Amount.String())
```

Comment:
nit: no need to change: wrong order of arguments. Here and others (t, expect, got)  (to be fair it is not consistent within `testify` like assert.Len(t, got, exp)` )